packages/edit-site/src/components/page-library/patterns-list.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * WordPress dependencies
+ */
+
+import {
+	SearchControl,
+	__experimentalHeading as Heading,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+	Flex,
+	FlexBlock,
+} from '@wordpress/components';
+import { __, isRTL } from '@wordpress/i18n';
+import { symbol, chevronLeft, chevronRight } from '@wordpress/icons';
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+import { useViewportMatch } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import Grid from './grid';
+import NoPatterns from './no-patterns';
+import usePatterns from './use-patterns';
+import SidebarButton from '../sidebar-button';
+import useDebouncedInput from '../../utils/use-debounced-input';
+import { unlock } from '../../lock-unlock';
+
+const { useLocation, useHistory } = unlock( routerPrivateApis );
+
+export default function PatternsList( { categoryId, label, type } ) {
+	const location = useLocation();
+	const history = useHistory();
+	const isMobileViewport = useViewportMatch( 'medium', '<' );
+	const [ filterValue, setFilterValue, delayedFilterValue ] =
+		useDebouncedInput( '' );
+
+	const [ patterns, isResolving ] = usePatterns(
+		type,
+		categoryId,
+		delayedFilterValue
+	);
+
+	const { syncedPatterns, unsyncedPatterns } = patterns;
+	const hasPatterns = !! syncedPatterns.length || !! unsyncedPatterns.length;
+
+	return (
+		<VStack spacing={ 6 }>
+			<Flex>
+				{ isMobileViewport && (
+					<SidebarButton
+						icon={ isRTL() ? chevronRight : chevronLeft }
+						label={ __( 'Back' ) }
+						onClick={ () => {
+							// Go back in history if we came from the library page.
+							// Otherwise push a stack onto the history.
+							if ( location.state?.backPath === '/library' ) {
+								history.back();
+							} else {
+								history.push( { path: '/library' } );
+							}
```

Comment:
> The reason we have to do this instead of leveraging NavigatorToParentButton is because this tree isn't wrapped in the NavigatorProvider context, nor is it under the /library subtree of the NavigatorScreen, so navigator.goBack() won't work here.

Original explanation: https://github.com/WordPress/gutenberg/pull/51558#discussion_r1231750792